src/Worker/UpdateGServers.php

```diff
@@ -47,7 +48,7 @@ public static function execute()
 		}
 
 		$total = DBA::count('gserver');
-		$condition = ["`next_contact` < UTC_TIMESTAMP() AND (`nurl` != ? OR `url` != ?)", '', ''];
+		$condition = ["(`nurl` != '' OR `url` != '') AND `next_contact` < ?", DateTimeFormat::utcNow()];
```

Comment:
Sorry for nitpicking, but `''` is a value - and in SQL queries we should always separate the values from the statements.

Reply:
Among values I would differentiate variables from constants. In this case, we will never need to change these two values to something different than empty string because we are checking that a value is set. The thing we should actually do is make both thee columns nullable and use a `IS NOT NULL` check instead of an empty string check or extracting empty string as parameter values.

Reply:
When we eventually will support multiple database engines like for example PostgreSQL we could come to the point where the one engine does need to have strings in `"` and the other in `'` - or there could be other side effects. Also the separation between content and query is a general rule when working with prepared statements.

Reply:
Ah, in that context, sure, but I still maintain the field should be nullable and we woudn't have had to have this discussion.